### PR TITLE
chore: enhace the SVG usage in visual editor

### LINKED ISSUE #1497: Use SVG in Visual Editor
## Describe the bug
Give a clear and concise description of what the bug is.

## Version
What version of the Composer are you using? Paste the build SHA found on the about page (`/about`).

## Browser
What browser are you using?

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS 
What operating system are you using?

- [ ] macOS
- [ ] Windows
- [ ] Ubuntu

## Expected behavior
Give a clear and concise description of what you expected to happen.

## Screenshots
If applicable, add screenshots/gif/video to help explain your problem.

### PR DESCRIPTION
**Pull request by yeze322**
## Description
closes #1497 
resovles #1906 

This PR improves the usage of SVG elements in VisualEditor, with the improvements, several pixel issues are fixed (#1906 ) and #1497 wil be closed.

## Details
Covered 3 areas of changes:
|-|Area|Description|
|----|----|-----|
|1|SVG semantic structure|[before] create a `<svg />` container per edge -> [after] multiple edges share one svg container|
|2|Fix pixel issues|fix the line cap/miter problem; fix the trailing circle not aligned issue; '+' icon size aligns to design|
|3|Implement the `EdgeUtil` lib|retire the old lib `EdgeComponent` which doesn't follows svg conventions|

**1. semantic svg**
- Multiple edges now share one `<svg />` container (compared to previous solution, create a new svg container for every edge).
  ```html
  // before
  <svg><line /></svg>
  <svg><line /></svg>
  
  // now
  <svg>
    <line />
    <line />
  </svg>
  ```
- The trailing circle is implemented as a `<circle />` element
  ```html
  // before
  <div style={{ borderRadius: '16px', border: '2px solid #979797' }} />
  // after
  <circle r="8" cx="50" cy="50" fill="none" stroke="#979797" strokeWidth="2" />
  ```
  It also fixes the circle alignment problem.
- An arrowhead is drawn by `<polyline />` instead of two lines

**2. fix pixel issues**
- fix the trailing circle alignment issue (by drawing it with svg.circle)
  [now]
  image
  [before]
  image
- fix the line cap/miter problem appears in an arrowhead (reported by William)
  [now]
  image
  [before]
  image
- Adds button on edge
image

**3. `EdgeUtil lib`**:
As a result of 1&2, this PR contains a new lib for drawing edges and retires the old one `EdgeComponent`.
image

image

## Task Item

#1497 
#1906

### CODE REVIEW

File: Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    strokeWidth: '1',
+    stroke: color,
+  };
+  const endPoint = calculateEdgeEndPoint(startPoint, direction, length);
+
+  const elements: JSX.Element[] = [];
+  // Draw the edge line
+  if (length > 0) {
+    const line = (
+      <line key={'edge__line'} x1={startPoint.x} y1={startPoint.y} x2={endPoint.x} y2={endPoint.y} {...strokeProps} />
```
Comment (compulim) [inline Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]:
```suggestion
      <line key="edge__line" x1={startPoint.x} y1={startPoint.y} x2={endPoint.x} y2={endPoint.y} {...strokeProps} />
```

File: Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    );
+    elements.push(line);
+  }
+  // Draw the edge label
+  if (typeof label === 'string' || typeof label === 'number') {
+    const text = (
+      <text
+        key={'edge__text'}
```
Comment (compulim) [inline Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]:
```suggestion
        key="edge__text"
```

File: Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        {label}
+      </text>
+    );
+    elements.push(text);
+  }
+  // Draw the edge arrow
+  if (arrowed) {
+    const arrowPoints = calculateArrowPoints(endPoint, direction);
+    const arrows = arrowPoints.map((p, index) => (
+      <line key={`edge__arrow-${index}`} x1={endPoint.x} y1={endPoint.y} x2={p.x} y2={p.y} {...strokeProps} />
```
Comment (compulim) [inline Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]: If the two lines of the arrow are drawn separately, there will be line cap/miter problems. So they are preferably joined as a single line.

http://www.bluevoid.com/opengl/sig00/advanced00/notes/node290.html

Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]: Thanks William! It's a really good advice

File: Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx (hunk 1 above)
Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]: Resolved. I created a commit manually to merge all this kind of changes to a single commit

File: Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx (hunk 3 above)
Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/components/lib/EdgeUtil.tsx]: resolved 
image

**Pull request merged**